### DISCUSSION TITLE
pnpm ship command error

### DISCUSSION THREAD

**Discussion by angela-yan** [netless-io/flat#1858] [Q&A]
Hi,

I tried to follow your instruction to build the app and had some error when I run 'pnpm ship' to for the electron app.

System: Win10
Source code: Download zip file into system

Error:
> flat@2.1.0 ship C:\Work\Projects\Collab\flat-main\desktop\main-app
> pnpm build && node ./scripts/pack auto

> flat@2.1.0 build C:\Work\Projects\Collab\flat-main\desktop\main-app
> cross-env NODE_ENV=production pnpm run _build:esbuild

> flat@2.1.0 _build:esbuild C:\Work\Projects\Collab\flat-main\desktop\main-app
> esbuild-dev --cjs scripts/esbuild/esbuild.prod.ts

**X [ERROR]** The define key "process.env.IntelliJ IDEA Community Edition" contains invalid identifier "IntelliJ IDEA Community Edition"

**X [ERROR]** The define key "process.env.IntelliJ IDEA Community Edition" contains invalid identifier "IntelliJ IDEA Community Edition"

 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
C:\Work\Projects\Collab\flat-main\desktop\main-app:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  flat@2.1.0 ship: `pnpm build && node ./scripts/pack auto`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.

**Comment by hyrious** [comment]
This is because the build script injects environment variables to esbuild's `define` and it didn't handle names with spaces in them. I can ignore this kind of names if it helps.

**Reply by hyrious** [reply]
Drafted a PR: https://github.com/netless-io/flat/pull/1859
